### PR TITLE
Updating to Pyprojecttoml, docker, tox, cloudbuild

### PR DESCRIPTION
**Pull request by bsieber-mozilla**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    { include = "pipeline" }
+]
+
+[[tool.poetry.source]]
+name = "mozilla-it"
+url = "https://dp2-prod.appspot.com/pypi"
+
+[tool.poetry.dependencies]
+python = ">3.6"
+tox = "*"
```
Comment (cvalaas) [inline pyproject.toml:25]: should this be a dev-dep?

Comment (bsieber-mozilla) [reply pyproject.toml:25]: It's a bit of a chicken and the egg problem in how we're testing with the docker; I imagine there's a better approach. 

But all dev deps can be listed in tox, so I don't know if anything should be in the dev-deps section.

Comment (bsieber-mozilla) [reply pyproject.toml:25]: This was also partially the reason I began to look into multiple docker files

Comment (bsieber-mozilla) [reply pyproject.toml:25]: (Trying a few things out with "extras" but if you get any luck with another approach or direction let me know!)

**Pull request merged**